src/ArborX_DBSCAN.hpp

```diff
@@ -149,15 +149,16 @@ dbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
   Kokkos::View<int *, MemorySpace> labels(
       Kokkos::ViewAllocateWithoutInitializing("ArborX::DBSCAN::labels"), n);
   ArborX::iota(exec_space, labels);
+  Details::UnionFind<MemorySpace> union_find{labels};
```

Comment:
Why do you want to expose this?  This is an implementation of the callback.  Take the labels like you did before and create your union find in the callback constructor.

Reply:
Well, I could do that in this PR, but I would have to move it out in the dense box PR, as I will also need to update this outside of the callback. So, rather than passing the labels around I like it better to pass union-find struct around. This also makes sense logically, as we could change the way union-find struct implemented (if desired) without changing the callback.